test/transaction.cpp

```diff
@@ -0,0 +1,163 @@
+    (at your option) any later version.
+
+    cpp-ethereum is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file transaction.cpp
+ * @author Christoph Jentzsch <cj@ethdev.com>
+ * @date 2014
+ * Transaaction test functions.
+ */
+
+#include <boost/filesystem/operations.hpp>
+#include <boost/test/unit_test.hpp>
+#include "JsonSpiritHeaders.h"
+#include <libdevcore/CommonIO.h>
+#include <libethereum/BlockChain.h>
+#include <libethereum/State.h>
+#include <libethereum/ExtVM.h>
+#include <libethereum/Defaults.h>
+#include <libevm/VM.h>
+#include "TestHelper.h"
+
+using namespace std;
+using namespace json_spirit;
+using namespace dev;
+using namespace dev::eth;
+using namespace dev::eth;
+
+namespace dev {  namespace test {
+
+bytes importByteArray(const std::string& str)
+{
+    bytes data;
+    if (str.find_first_of("0x") == 0)
+            data = fromHex(str.substr(2));
+        else
+            data = fromHex(str);
+    return data;
+}
+
+void doTransactionTests(json_spirit::mValue& v, bool _fillin)
+{
+    processCommandLineOptions();
+
+    for (auto& i: v.get_obj())
+    {
+        cerr << i.first << endl;
+        mObject& o = i.second.get_obj();
+
+        BOOST_REQUIRE(o.count("transaction") > 0);
+
+        mObject tObj = o["transaction"].get_obj();
+
+        BOOST_REQUIRE(tObj.count("data") > 0);
+        BOOST_REQUIRE(tObj.count("value") > 0);
+        BOOST_REQUIRE(tObj.count("gasPrice") > 0);
+        BOOST_REQUIRE(tObj.count("gasLimit") > 0);
+        BOOST_REQUIRE(tObj.count("nonce")> 0);
+        BOOST_REQUIRE(tObj.count("to") > 0);
+
+        BOOST_REQUIRE(tObj.count("v") > 0);
+        BOOST_REQUIRE(tObj.count("r") > 0);
+        BOOST_REQUIRE(tObj.count("s") > 0);
+
+        //Construct Rlp of the given transaction
+        RLPStream rlpStream;
+        rlpStream.appendList(9);
+        rlpStream << toInt(tObj["nonce"]) << toInt(tObj["gasPrice"]) << toInt(tObj["gasLimit"]);
+        if (tObj["to"].get_str().empty())
+            rlpStream << "";
+        else
+            rlpStream << Address(tObj["to"].get_str());
+        rlpStream << toInt(tObj["value"]) << importData(tObj);
+
+        u256 r = h256(fromHex(tObj["r"].get_str()));
+        u256 s = h256(fromHex(tObj["s"].get_str()));
+        rlpStream << (toInt(tObj["v"]) + 27) << r << s;
+
+        Transaction tr(rlpStream.out(), true);
+        if(_fillin)
+        {
+               o["rlp"] = "0x" + toHex(tr.rlp());
+               o["sender"] = toString(tr.sender());
+        }
+
+        BOOST_REQUIRE(o.count("rlp") > 0);
+        BOOST_REQUIRE(o.count("sender") > 0);
```

Comment:
you cant require sender. In an invalid transaction thi object is not part of the tests (see spec.)